### ISSUE TITLE
Can't log in with webkit.

### ISSUE THREAD

**Issue by rplatel** [zaidka/genieacs-gui#64]
When attempting to log in with Chrome or Safari I get a ActionController::InvalidAuthenticityToken error.  I can successfully log in with Firefox.

I believe the issue is that the login page contains two csrf authenticity tokens, one in the <head>:

```
<meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="5/WxVZ1x8YHIlLCrZytBvNBlEmE8ipfl/9VdG2c9lhn4rZfg7VhegHaSOc1QG9a9ut2U1xyt7jWT0at7Lm55ZQ==" />

```
and a different one in the login form
```
<input type="hidden" name="authenticity_token" value="KzrDjf6MiuraXa/zVPpa7wnSk6DPprZXDe5RcQhO2sQFHtl2+wOuTVUjJ1ahXY9th1nNJCPiNSLv/IPUmKZFuw==" />
```

tcpdumping, I see that Firefox sends the token from the meta tag in the POST, but Safari sends the token from the form.

Editing new.html.erb to remove the token from the form with
```<%= form_tag log_in_path, :authenticity_token => false do %> ```
does remove the token from the form, but then Safari sends no token and I get the same error.

**Comment by zaidka** [comment]
Those authenticity tokens are automatically inserted by the Rails framework. Try updating your gems and see if the issue persists. Note that I haven't tested genieacs-gui with Rails 5 so if you're using Rails 5 then try downgrading to 4.x.

**Comment by rplatel** [comment]
The Gemfile and Gemfile.lock on github seem to be specifying rails 5.0.0, so that's what we ended up with (I am not a rails or gem expert)

```
DEPENDENCIES
  byebug
  coffee-rails (~> 4.2)
  jbuilder (~> 2.5)
  jquery-rails
  json
  listen (~> 3.0.5)
  puma (~> 3.0)
  rails (~> 5.0.0)
  sass-rails (~> 5.0)
  spring
  spring-watcher-listen (~> 2.0.0)
  sqlite3
  turbolinks (~> 5)
  tzinfo-data
  uglifier (>= 1.3.0)
  web-console
```

**Comment by zaidka** [comment]
You're right. I think I was confusing 5.0 with the recently released 5.1.

I suspect the error has something to do with Turbolinks gem. Try disabling it as described here and let me know if that changes anything: http://stackoverflow.com/questions/38649550/how-to-disable-turbolinks-in-rails-5/38649595#38649595

**Comment by rplatel** [comment]
Disabling turbolinks didn't work for us.  
Turning off CSRF protection in application_controller.rb with
```skip_before_action :verify_authenticity_token```
works, but obviously is not ideal.

**Comment by hurrycaner** [comment]
Check this rails issue: https://github.com/rails/rails/issues/21948

**Comment by Kalpar** [comment]
Figure I would update an old issue as I had the same error, but it was because I was behind a Nginx proxy. The final solution was to add these additional reverse proxy headers to the Nginx config:

```
server {
        ...
        location / {
                ....
                proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header  X-Forwarded-Proto $scheme;
                proxy_set_header  X-Forwarded-Ssl on; # Optional
                proxy_set_header  X-Forwarded-Port $server_port;
                proxy_set_header  X-Forwarded-Host $host;
        }
}
```
